helper-cli/src/main/kotlin/commands/scanstorage/DeleteCommand.kt

```diff
@@ -0,0 +1,122 @@
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.helper.commands.scanstorage
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.options.associate
+import com.github.ajalt.clikt.parameters.options.convert
+import com.github.ajalt.clikt.parameters.options.default
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.types.enum
+import com.github.ajalt.clikt.parameters.types.file
+
+import org.jetbrains.exposed.sql.Database
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.isNotNull
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.like
+import org.jetbrains.exposed.sql.compoundAnd
+import org.jetbrains.exposed.sql.deleteWhere
+
+import org.ossreviewtoolkit.helper.common.ORTH_NAME
+import org.ossreviewtoolkit.model.SourceCodeOrigin
+import org.ossreviewtoolkit.model.config.OrtConfiguration
+import org.ossreviewtoolkit.model.config.PostgresStorageConfiguration
+import org.ossreviewtoolkit.model.utils.DatabaseUtils
+import org.ossreviewtoolkit.model.utils.DatabaseUtils.transaction
+import org.ossreviewtoolkit.model.utils.rawParam
+import org.ossreviewtoolkit.scanner.storages.utils.ScanResults
+import org.ossreviewtoolkit.utils.ORT_CONFIG_FILENAME
+import org.ossreviewtoolkit.utils.expandTilde
+import org.ossreviewtoolkit.utils.log
+import org.ossreviewtoolkit.utils.ortConfigDirectory
+
+internal class DeleteCommand : CliktCommand(
+    help = "Removes stored scan results matching the options or all results if no options are given."
+) {
+    private val configFile by option(
+        "--config",
+        help = "The path to the ORT configuration file that configures the scan results storage."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .default(ortConfigDirectory.resolve(ORT_CONFIG_FILENAME))
+
+    private val configArguments by option(
+        "-P",
+        help = "Override a key-value pair in the configuration file. For example: " +
+                "-P scanner.postgresStorage.schema=testSchema"
+    ).associate()
+
+    private val provenance by option(
+        "--provenance",
+        help = "The origin of the scan results that should be deleted."
+    ).enum<SourceCodeOrigin>()
+
+    private val type by option(
+        "--type",
+        help = "The package manager type to delete from the scan storage. Like 'Maven' or 'NPM'."
+    )
+
+    override fun run() {
+        val database = initDatabase(OrtConfiguration.load(configArguments, configFile))
+
+        val provenanceKey = when (provenance) {
+            SourceCodeOrigin.ARTIFACT -> "source_artifact"
+            SourceCodeOrigin.VCS -> "vcs_info"
+            null -> null
```

Comment:
I believe making the `provenance` required, and turning it into a collection (created from CSV) would simplify the logic here. What do you think?

Reply:
I like the suggestion. Please have a look at the changes.